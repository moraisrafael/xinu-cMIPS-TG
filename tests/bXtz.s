	.include "cMIPS.s"
	.text
	.align 2
	.set noat
	.globl _start
	.ent _start
_start: nop
	la $15,x_IO_BASE_ADDR
	li $3,1
	li $4,5
	li $1,-6
	move $8,$zero
	nop
snd:	add  $8,$8,$3    # $8 <-  1, 7,13,19,25,31,
	add  $8,$8,$4    # $8 <-  6,12,18,24,30,36,
	add  $9,$8,$8    # $9 <- 12,24,36,48,60,72,
	sw   $9, 0($15)
	addiu $1,$1,1
	nop
	bltz $1,snd	#12,24,36,48,60,72      $1:=-6,addi+1
	nop
	sw   $0, 4($15)
	li $1,-6
snd2:	add  $8,$8,$3    # $8 <-  1, 7,13,19,25,31,
	add  $8,$8,$4    # $8 <-  6,12,18,24,30,36,
	add  $9,$8,$8    # $9 <- 12,24,36,48,60,72,
	sw   $9, 0($15)
	addiu $1,$1,1
	nop
	blez $1,snd2	#12,24,36,48,60,72,84   $1:=-6,addi+1
	nop
	sw   $0, 4($15)
	li $1,6
snd3:	add  $8,$8,$3    # $8 <-  1, 7,13,19,25,31,
	add  $8,$8,$4    # $8 <-  6,12,18,24,30,36,
	add  $9,$8,$8    # $9 <- 12,24,36,48,60,72,
	sw   $9, 0($15)
	addiu $1,$1,-1
	nop
	bgez $1,snd3	#12,24,36,48,60,72,84   $1:=6,addi-1
	nop
	sw   $0, 4($15)
	li $1,6
snd4:	add  $8,$8,$3    # $8 <-  1, 7,13,19,25,31,
	add  $8,$8,$4    # $8 <-  6,12,18,24,30,36,
	add  $9,$8,$8    # $9 <- 12,24,36,48,60,72,
	sw   $9, 0($15)
	addiu $1,$1,-1
	nop
	bgtz $1,snd4	#12,24,36,48,60,72      $1:=6,addi-1
	nop
	sw   $0, 4($15)
	nop
	nop
	nop
	wait
	nop
	.end _start
