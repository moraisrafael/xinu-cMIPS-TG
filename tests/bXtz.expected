0000000c
00000018
00000024
00000030
0000003c
00000048
00000000
00000054
00000060
0000006c
00000078
00000084
00000090
0000009c
00000000
000000a8
000000b4
000000c0
000000cc
000000d8
000000e4
000000f0
00000000
000000fc
00000108
00000114
00000120
0000012c
00000138
00000000
